pyuvdata/utils.py

```diff
@@ -1081,9 +1081,64 @@ def get_lst_for_time(jd_array, latitude, longitude, altitude):
     return lst_array
 
 
-def find_clusters(location_ids, location_vectors, tol):
+def _adj_list(vecs, tol, n_blocks=None):
+    """Identify neighbors of each vec in vecs, to distance tol."""
+    n_items = len(vecs)
+    max_items = 2 ** 10  # Max array size used is max_items**2. Avoid using > 1 GiB
+
+    if n_blocks is None:
+        n_blocks = max(n_items // max_items, 1)
+
+    # We may sort blocks so that some pairs of blocks may be skipped.
+    # Reorder vectors by x.
+
+    order = np.argsort(vecs[:, 0])
+    blocks = np.array_split(order, n_blocks)
+    adj = [{k} for k in range(n_items)]  # Adjacency lists
+    for b1 in blocks:
+        for b2 in blocks:
+            v1, v2 = vecs[b1], vecs[b2]
+            # Check for no overlap, with tolerance.
+            xmin1 = v1[0, 0] - tol
+            xmax1 = v1[-1, 0] + tol
+            xmin2 = v2[0, 0] - tol
+            xmax2 = v2[-1, 0] + tol
+            if max(xmin1, xmin2) > min(xmax1, xmax2):
+                continue
+
+            adj_mat = cdist(vecs[b1], vecs[b2]) < tol
+            for bi, col in enumerate(adj_mat):
+                adj[b1[bi]] = adj[b1[bi]].union(b2[col])
+    return [frozenset(g) for g in adj]
+
+
+def _find_cliques(adj, strict=False):
+    n_items = len(adj)
+
+    loc_gps = []
+    visited = np.zeros(n_items, dtype=bool)
+    for k in range(n_items):
+        if visited[k]:
+            continue
+        a0 = adj[k]
+        visited[k] = True
+        if all(adj[it].__hash__() == a0.__hash__() for it in a0):
+            group = list(a0)
+            group.sort()
+            visited[list(a0)] = True
+            loc_gps.append(group)
+
+    # Require all adjacency lists to be isolated maximal cliques:
+    if strict:
+        if not all(sorted(st) in loc_gps for st in adj):
+            raise ValueError("Non-isolated cliques found in graph.")
+
+    return loc_gps
+
+
+def find_clusters(location_ids, location_vectors, tol, strict=False):
```

Comment:
Please add the new `strict` parameter to the Parameters section of the docstring.

Reply:
Whoops, thanks!